packages/flutter_tools/lib/src/base/dds.dart

```diff
@@ -57,6 +58,11 @@ class DartDevelopmentService {
       logger.printTrace('DDS is listening at ${_ddsInstance.uri}.');
     } on dds.DartDevelopmentServiceException catch (e) {
       logger.printTrace('Warning: Failed to start DDS: ${e.message}');
+      if (e.errorCode == dds.DartDevelopmentServiceException.existingDdsInstanceError) {
+        _existingDdsUri = Uri.parse(
```

Comment:
As a cleanup it would be nice to provide this as an optional field on the DartDevelopmentServiceException